### ISSUE TITLE
Reduce visual regression test time

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#1354]
I'd like to write a list of possible tasks to reduce the time for visual regression tests:

## Reduce the time spent on generating server required files
~30 seconds are spent on generating the required pages and files for the VRTs server.
Most of the time is spent creating the `examples.json` file that basically represents the list of all the existing storybook stories. This file is created by going and compiling each story with Typescript and extracting info like filename, exported story name, and the group name.
We can reduce this time to near zero if we simplify this extraction by
- [ ] Define and a descriptive comment to each story. This will help now and in the future defining a better naming and grouping for stories that doesn't depend on the file name or the exported name
- [ ] Build an eslint plugin that requests the descriptive comment or fixit automatically if missing.
- [ ] Refactor the generator to read files as text (instead of compiling them) and grep the required comment, extracting the required info

Pros:
- we don't have to change the way we declare each story, we just need to add, on the same file, few more lines to describe the story. Storybook will continue to work.
- The time to compile the `examples.json` file should drops because we just read the file as text without any compilation
- we can reuse this also outside Storybook if we are going to move away from it
- we can use this to simplify the grouping/naming of each story

## Reduce server bootstrap time
The VRT server right now takes nearly 1 minute to boot (~2 on CI). 
It is caused by: build the project, compile SASS to CSS
The compilation builds a very large number of chunks to reduce the size of the loaded bundle when the VRT test navigate to a new Story page. 
We should try multiple ways:
- [ ] try to build a single bundle, loading and keeping it in Puppeteer cache. It really depends on the size of this bundle and the time to parse/compile such a bigger js bundle
- [ ] reduce ad limit the libraries used and compiled by the server. The log shows that we compile and emit chunks of every icon that exists in EUI.
- [ ] use `esbuild-loader` to improve the build speed. I've tested locally and it took nearly 8/10 seconds to compile, instead of the 30/40 seconds with TSC.
- [ ] parallelize the SASS to CSS compilation and load them through the HTML page instead of through react/webpack
- [ ] Try to remove the Terser plugin, the optimization steps and any unnecessary plugins.
- [ ] extra: a file change recompiles completely the project, taking another 40 seconds, we should improve this to provide incremental builds if you are running the server for dev purposes.

## Reduce the Docker-Chromium image build

- [ ] on a local environment we keep the cache of the last built image, but on CI and in particular, on Jenkins, this cache is not preserved. We should contact Infra to ask if we can preserve the cache for that docker image.

## Parallelize current test set

- [x] We currently run 2 parallel set of tests: one that runs all the available Storybook stories (`integration/tests/all.test.ts`), and another that runs all the specific tests written in the `integration/tests` folder (except the `all.test.ts`). We can improve the parallelization by splitting these sets into half or smaller parts. This parallelization should run only after having built the static HTML files (right now the two parallel workers executes the same setup steps, from npm install to building the app, that is a waist of CI time)
- [x] parallelize tests on the same test suite: each test is currently executed in a sequential manner, a single page is used to take one screenshot at time. We can probably make this parallelizable by creating a test page for each test suite some example/ideas [jest](https://stackoverflow.com/questions/49934994/parallel-support-using-jest-with-puppeteer) [puppeteer cluster](https://github.com/thomasdondorf/puppeteer-cluster) [playwright](https://playwright.dev/docs/test-parallel)
- [x] Try again Playwright. I've tested it multiple times, the test infrastructure works way better (you can see each test reported when completed, you can probably add the test coverage) but it looks like the screenshot diffing took a bit longer then our tests, nearly 2x the time we spent on screenshotting and diffing.

## Cleanup and the improve load time of each single story
- [ ] The current time spent on loading the page, taking the screenshot, and diffing is nearly 800ms and can vary from the complexity of the example. With https://github.com/elastic/elastic-charts/pull/1366 I'm trying to simplify and reduce the time by avoiding anything that is not necessary (eg: eui icon and any particular EUI side effect loading)
- [ ] We can probably use the History API to switch between examples. This should reduce the loading time because with the history we just have to load the next js chunk instead of loading everytime the main bundle when using `page.goto`.

**Comment by markov00** [comment]
I think we can pause this effort now, the VRT server seems to serves very well right now, so all improvements can be paused
